d3a_api_client/commands.py

```diff
@@ -0,0 +1,68 @@
+from abc import ABC, abstractmethod
+
+from d3a_api_client.enums import Commands, command_enum_to_command_name
+
+
+class Command(ABC):
+    @abstractmethod
+    def execute(self):
+        pass
+
+
+class ClientCommand(Command):
+
+    def __init__(self, area_uuid):
+        super(ClientCommand, self).__init__()
+        self.area_uuid = area_uuid
+        self.action = None
+        self.callback = None
+        self.callback_args = {}
+
+    def offer_energy(self, energy, price):
+        self.callback_args = {energy, price}
+        self.action = Commands.OFFER
+
+    def offer_energy_rate(self, energy, rate):
+        self.callback_args = {energy, rate}
+        self.action = Commands.OFFER
+
+    def bid_energy(self, energy, rate):
+        self.callback_args = {energy, rate}
+        self.action = Commands.OFFER
+
+    def delete_offer(self, offer_id):
+        self.callback_args = {"offer_id": offer_id}
+        self.action = Commands.DELETE_OFFER
+
+    def delete_bid(self, bid_id):
+        self.callback_args = {"bid_id": bid_id}
+        self.action = Commands.DELETE_OFFER
+
+    def list_offers(self):
+        self.action = Commands.LIST_OFFERS
+
+    def list_bids(self):
+        self.action = Commands.LIST_BIDS
+
+    def device_info(self):
+        self.action = Commands.DEVICE_INFO
+
+    def execute(self) -> dict:
+        if self.area_uuid:
+            return {self.area_uuid: {"type": command_enum_to_command_name(self.action), **self.callback_args}}
+
+
+class ClientCommandList(Command, list):
```

Comment:
There is no need for multiple inheritance and extending the list here. You can have a private member, namely `_pending_commands`, that will hold all pending commands that have not been sent yet. 
Generally speaking, I usually favor composition over inheritance: https://en.wikipedia.org/wiki/Composition_over_inheritance

Reply:
I have this already implemented as `commands_buffer` variable in the RedisClient class , I'll make it private and make it a list instead of a ClientCommandList object , the only issue here is where to put the batch commands execute logic , I can make it a static function inside the ClientCommand class if suitable

Reply:
Instead of keeping the container on the RedisClient class, why not storing it in the ClientCommand class directly ? That way you would store and control the functionality of the commands in the same class, and also you would not have to make the `execute()` method static.

Reply:
@spyrostz Done , can you check now please

Reply:
That looks good, thanks! I have left some other comments for renaming the methods to correctly indicate their intent.